### PR TITLE
Implement LFU evictor for client cache

### PR DESCRIPTION
**Pull request by bf8086**

### CODE REVIEW

File: core/client/fs/src/main/java/alluxio/client/file/cache/evictor/LFUCacheEvictor.java [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+    mLogBase = conf.getDouble(PropertyKey.USER_CLIENT_CACHE_EVICTOR_LFU_LOGBASE);
+  }
+
+  private int getBucket(int count) {
+    return (int) (Math.log(count) / Math.log(mLogBase));
+  }
+
+  private void addPageToBucket(PageId pageId, int bucket) {
+    mBucketMap.compute(bucket, (bucketKey, pageSet) -> {
+      Set<PageId> set = pageSet == null ? new LinkedHashSet<>() : pageSet;
```
Comment (apc999) [inline core/client/fs/src/main/java/alluxio/client/file/cache/evictor/LFUCacheEvictor.java]: If we change `LinkedHashSet` to `LinkedHashMap` with access order (check `LRUCacheEvictor`), don't we have a LRU bucket in that case? @calvinjia

File: core/client/fs/src/main/java/alluxio/client/file/cache/evictor/LRUCacheEvictor.java [hunk 2]
```diff
@@ -36,6 +37,14 @@
       Collections.synchronizedMap(new LinkedHashMap<>(LINKED_HASH_MAP_INIT_CAPACITY,
           LINKED_HASH_MAP_INIT_LOAD_FACTOR, LINKED_HASH_MAP_ACCESS_ORDERED));
 
+  /**
+   * Required constructor.
+   *
+   * @param conf Alluxio configuration
+   */
+  public LRUCacheEvictor(AlluxioConfiguration conf) {
```
Comment (apc999) [inline core/client/fs/src/main/java/alluxio/client/file/cache/evictor/LRUCacheEvictor.java]: oops

File: core/client/fs/src/main/java/alluxio/client/file/cache/evictor/LFUCacheEvictor.java (hunk 1 above)
Comment (bf8086) [reply core/client/fs/src/main/java/alluxio/client/file/cache/evictor/LFUCacheEvictor.java]: Done.

File: core/client/fs/src/main/java/alluxio/client/file/cache/evictor/LFUCacheEvictor.java [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+   * Required constructor.
+   *
+   * @param conf Alluxio configuration
+   */
+  public LFUCacheEvictor(AlluxioConfiguration conf) {
+    mLogBase = conf.getDouble(PropertyKey.USER_CLIENT_CACHE_EVICTOR_LFU_LOGBASE);
+  }
+
+  private int getBucket(int count) {
+    return (int) (Math.log(count) / Math.log(mLogBase));
```
[the remark this answers is not in the document]
Comment (bf8086) [reply core/client/fs/src/main/java/alluxio/client/file/cache/evictor/LFUCacheEvictor.java]: Good idea! Updated.

**Comment by bf8086** [comment]
alluxio-bot, test this please.

**Comment by bf8086** [comment]
alluxio-bot, merge this please.

**Comment by bf8086** [comment]
alluxio-bot, cherry-pick to branch-2.2 please.

**Pull request merged**